### ISSUE TITLE
`atexit.register` does not work

### ISSUE THREAD

**Issue by clubby789** [Gallopsled/pwntools#1850]
### What I did
```python3
from pwn import *
atexit.register(print, "hello world")
exit()
```
### What I expected to see
```python3 test.py
hello world
```
### What I saw
Nothing

I noticed this because `asm()`, which adds an `atexit` handler to remove the `/tmp/pwn-asm-XXXXXX` folder, does not in fact remove it, meaning multiple script runs leads to many similar folders.

**Comment by clubby789** [comment]
Looks like https://github.com/Gallopsled/pwntools/blob/ec9dfe108b85ac47983e9a98808fcdbb50cb0cdd/pwnlib/atexit.py#L98 is to blame, as `sys.exitfunc` has been removed in Python 3.X (https://docs.python.org/3.5/search.html?q=sys.exitfunc&check_keywords=yes&area=default), encouraging the use of `atexit` (the stdlib module) instead.

**Comment by Arusekk** [comment]
GH forgot to mark this as closed in #1852
